Tests/Reqnroll.SystemTests/Generation/GenerationTestBase.cs

```diff
@@ -443,5 +443,101 @@ public async Task WhenThisTableIsProcessed(DataTable table)
 
     #endregion
 
-    //TODO: test parallel execution (details TBD) - maybe this should be in a separate test class
+    #region It is able to run tests parallel
+
+    // We need to verify
+    // Tests are run in parallel
+    // Each scenario execution has a feature context and scenario context that is matching to the feature/scenario
+    // For each scenario the before feature has been executed for that feature context
+
+    [TestMethod]
+    public void Tests_can_be_executed_parallel()
+    {
+        _projectsDriver.EnableTestParallelExecution();
+
+        var scenarioTemplate = 
+            """
+            Scenario: {0}
+              When executing '{0}' in '{1}'
+            """;
+
+        var scenarioOutlineTemplate = 
+            """
+            Scenario Outline: {0}
+              When executing '{0}' in '{1}'
+            Examples:
+            	| Count |
+            	| 1     |
+            	| 2     |
+                | 3     |
+            """;
+
+        const int scenariosPerFile = 3;
+        const int scenarioOutlinesPerFile = 3;
+        string[] features = ["A", "B"];
+        int scenarioIdCounter = 0;
+
+        foreach (string feature in features)
+        {
+            AddFeatureFile($"Feature: {feature}" + Environment.NewLine);
+            for (int i = 0; i < scenariosPerFile; i++)
+                AddScenario(string.Format(scenarioTemplate, $"S{++scenarioIdCounter}", feature));
+            for (int i = 0; i < scenarioOutlinesPerFile; i++)
+                AddScenario(string.Format(scenarioOutlineTemplate, $"S{++scenarioIdCounter}", feature));
+        }
+
+        AddBindingClass(
+            """
+            namespace ParallelExecution.StepDefinitions
+            {
+                [Binding]
+                public class ParallelExecutionSteps
+                {
+                    public static int startIndex = 0;
+                    
+                    private readonly FeatureContext _featureContext;
+                    private readonly ScenarioContext _scenarioContext;
+                    
+                    public ParallelExecutionSteps(FeatureContext featureContext, ScenarioContext scenarioContext)
+                    {
+                        _featureContext = featureContext;
+                        _scenarioContext = scenarioContext;
+                    }
+                    
+                    [BeforeFeature]
+                    public static void BeforeFeature(FeatureContext featureContext)
+                    {
+                        featureContext.Set(true, "before_feature");
+                    }
+                
+                    [When("executing {string} in {string}")]
+                    public void WhenExecuting(string scenarioName, string featureName)
+                    {
+                        var currentStartIndex = System.Threading.Interlocked.Increment(ref startIndex);
+                        global::Log.LogStep();
```

Comment:
On my machine `Log` sometimes fails because the log files are locked more than 5 times by another thread.
Maybe we need to increase the retry count with all this parallel execution? 😁

Reply:
@obligaron Thx for the feedback. Maybe it's not the "5 times" we should increase, but use a random wait instead of the fixed `Thread.Sleep(500);` in the log code. I will make some experiments with it and let you know.

Reply:
I run the tests multiple times locally and didn't encountered the error anymore. Looks good to me. 🙂